### PR TITLE
Custom Permission Field

### LINKED ISSUE #29: Create PermissionSetField
**Is your feature request related to a problem? Please describe.**
We need a custom django model field to store our permissions as string and retieve it as python object as per our db design.
https://dbdiagram.io/d/5e97237739d18f5553fda63b

**Describe the solution you'd like**
1. Create a custom django model field which can automatically convert permissions into string for storing in the db and string into python object for using.
1. Place it in the org app
1. Write testcases to ensure proper functionality

### PR DESCRIPTION
**Pull request by naveennvrgup**
### Description
1. Created PermissionSet class
1. Created PermissionField class
1. Added tests for both classes
1. Added migrations for PermissionSetTestModel
1. Verified proper functioning of PermissionField

**Credits:**
1. @BadduCoder - suggested the idea of using bit-manipulation instead of using string
1. @priyanshisharma - for efforts in #40 
1. @sahil9001  - for efforts in #39 

**Type of Change:** (Delete irrelevant options)
- Code
- Testing
- Documentation

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

### Checklist:
- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by naveennvrgup** [comment]
@priyanshisharma  @sahil9001 Please go through this PR and check for errors. Submit a review if possible. You both made great efforts for #29 . :+1:

**Review by naveennvrgup — COMMENTED** [review]
The tests are moved to a common test folder.

File: tests/utils/tests_permission_field.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        self.assertTrue(record_from_db.perm.permissions[PermissionSet.CAN_REPLY_TO_QUERIES])
+        self.assertTrue(record_from_db.perm.permissions[PermissionSet.CAN_CREATE_TASKS])
+
+        self.assertFalse(record_from_db.perm.permissions[PermissionSet.IS_STAFF])
+        self.assertFalse(record_from_db.perm.permissions[PermissionSet.CAN_REVIEW_PROOFS])
+
+
+    def test_multiple_record_creation(self):
+        # create a permission set records
+        p_one = 1 << PermissionSet.IS_ADMIN
```
Comment (sumit-badsara) [inline tests/utils/tests_permission_field.py:114]: Better to name the variables as permission name or something related.

**Review by sumit-badsara — CHANGES_REQUESTED** [review]
Suggestion to improve readability of code ! Other things which I suggested (Using loops instead of hard coding) can be improved later on as an improvement PR.

**Review by sumit-badsara — APPROVED** [review]

**Pull request merged**